The ol' spicy keychain

BarneyGumball opened the pull request:
Let's you heat up pretty much any holdable item on the bunsen burner and in both the oven and deep fryer.  The item will then burn whoever proceeds to pick it up without the appropriate protection.  That protection being a pair of highly heat proof gloves.

![](http://i.imgur.com/LUTKCX2.png)

:cl:
 * rscadd: You can now heat up items on the bunsen burner and in both the oven and deep fryer to play hilarious pranks on your friends!  Be careful though, whatever you're holding will get really hot.  Best you wear something to protect your hands.

Shadowmech88 commented:
The addition of a new item-level var solely for a meme that can only be performed with bunsen burners seems odd.
I'd rather have a var to properly track the item's actual temperature a la reagent heating, along with a var to determine how quickly it gains/loses heat based on the type of material from which it is made.

BarneyGumball commented:
I can move the var to an item flag.

I don't like the idea of item temperatures though.  As it is, heating an item functions as a quick little "haha" moment, at which point you can go back to using the item.  Having proper item heating/cooling has much longer lasting implications for any items that get heated up.  I can only imagine the nightmare after a plasma fire.  Also, having every `/obj` in the world process temperature changes probably wouldn't be worth the performance cost.

I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE commented:
I imagine the best way performance-wise to implement this would be an on-pickup event but I'm still not convinced adding a single var is that bad

Also, what if spicy items had a red outline like staff of reanimation mobs?

kilozombie commented:
> I imagine the best way performance-wise to implement this would be an on-pickup event but I'm still not convinced adding a single var is that bad
> 
> Also, what if spicy items had a red outline like staff of reanimation mobs?

There would have to be a tiny window where it's spicy hot but not visibly hot so you can really give somebody the ol' spicy cryptographic sequencer

I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE commented:
Excuse me in the original spicy keychain picture the keys are visibly red

BarneyGumball commented:
>don't make it just a bunsen burner

I'm open to suggestions.

kilozombie commented:
Have it be a function in the oven, like in the original

BarneyGumball commented:
Alright, so I:
* Adjusted the contents of some messages.
* Made it so you can heat up items in the oven and the deep fryer.
* Fixed the gloves protection so you can't just remove your gloves after you've picked up the heated item.

BarneyGumball commented on code/game/objects/items.dm:
```diff
@@ -807,6 +810,53 @@
+				user.u_equip(src, 1)
+			user.audible_scream()
+			user.visible_message(
+					"<span class='warning'>[user] cries out in pain and jerks \his [hand_used] \
```
I don't actually know if `[hand_used]` works here, pls advise.

Edit: Ok it should work.  Not sure how to test it on my own though.

MadmanMartian commented:
So, added a proc for you to use. Attempt_heating. It's object level, takes an argument of the object being used to heat it, and the user if available.

Use that.

BarneyGumball commented:
This pr SUCKS
